systemverilog-plugin/UhdmAst.cc

```diff
@@ -1902,33 +1902,22 @@ void UhdmAst::process_module()
     } else {
         // Not a top module, create instance
         current_node = make_ast_node(AST::AST_CELL);
-        std::string module_parameters;
+        std::vector<std::pair<RTLIL::IdString, RTLIL::Const>> parameters;
         visit_one_to_many({vpiParamAssign}, obj_h, [&](AST::AstNode *node) {
             if (node && node->type == AST::AST_PARAMETER) {
                 if (node->children[0]->type != AST::AST_CONSTANT) {
                     if (shared.top_nodes.count(type)) {
                         simplify_parameter(node, shared.top_nodes[type]);
-                        log_assert(node->children[0]->type == AST::AST_CONSTANT || node->children[0]->type == AST::AST_REALVALUE);
+                    } else {
+                        simplify_parameter(node, nullptr);
```

Comment:
What do we expect from parameter simplification in case where there is no other node of the same type in `shared.top_nodes`?

Reply:
Sometimes we don't need access to cell instance to simplify parameter, e.g. in case of primitive cell and parameter that uses concat. We are expecting ``simplify_parameter`` to reduce this concat to constant value.

Reply:
I see, thanks!